src/codegen_internal.h

```diff
@@ -1,6 +1,6 @@
 // This file is a part of Julia. License is MIT: http://julialang.org/license
 
-#if defined(LLVM38) && !defined(LLVM37)
+#if defined(USE_ORCJIT) && JL_LLVM_VERSION <= 30800
```

Comment:
I think @yuyichao specifically requested this change: https://github.com/JuliaLang/julia/pull/17826#discussion_r73792427